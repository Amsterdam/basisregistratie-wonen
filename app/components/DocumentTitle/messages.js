import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  default_title: {
    id: `${scope}.registration_plural`,
  },
});
