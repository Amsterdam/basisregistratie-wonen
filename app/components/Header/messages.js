/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  logged_in_as: {
    id: `${scope}.logged_in_as`,
  },
  log_in: {
    id: `${scope}.log_in`,
  },
  log_in_adw: {
    id: `${scope}.log_in_adw`,
  },
  log_out: {
    id: `${scope}.log_out`,
  },
});
