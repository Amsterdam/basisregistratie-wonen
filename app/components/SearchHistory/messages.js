/*
 * SearchHistory Messages
 *
 * This contains all the text for the SearchHistory component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  search_history_header: {
    id: `${scope}.search_history_header`,
    defaultMessage: 'Recently search for',
  },
});
