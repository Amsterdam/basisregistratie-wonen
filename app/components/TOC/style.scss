.has-focus {
  outline: 4px solid #ffc90e;
  margin: 0;
  padding: 0 0 0 15px;
  background-color: #ffc90e !important;
  color: #000 !important;
}

.links li {
  line-height: 22px;
}

.links li > a {
  padding-left: 12px;
}
