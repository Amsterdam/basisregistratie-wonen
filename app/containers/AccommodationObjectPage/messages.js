/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  yes: {
    id: 'boilerplate.containers.AccommodationObjectPage.yes',
    defaultMessage: 'Ja',
  },
  no: {
    id: 'boilerplate.containers.AccommodationObjectPage.no',
    defaultMessage: 'Nee',
  },
  does_not_apply: {
    id: 'boilerplate.containers.AccommodationObjectPage.does_not_apply',
    defaultMessage: 'n.v.t.',
  },
});
