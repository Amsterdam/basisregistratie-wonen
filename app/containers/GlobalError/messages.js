/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  unable_to_fetch: {
    id: `${scope}.unable_to_fetch`,
  },
  unauthorized: {
    id: `${scope}.unauthorized`,
  },
  session_expired: {
    id: `${scope}.session_expired`,
  },
});
