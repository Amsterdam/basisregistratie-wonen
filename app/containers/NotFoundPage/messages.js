/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  document_title: {
    id: `${scope}.registration_plural`,
  },
  header: {
    id: `${scope}.page_not_found`,
  },
});
