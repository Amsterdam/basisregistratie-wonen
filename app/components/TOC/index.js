import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { intlShape } from 'react-intl';

import SectionHeading from 'components/SectionHeading';
import messages from 'containers/App/messages';
import Link from 'components/Link';
import LoadingIndicator from 'components/LoadingIndicator';

const Li = styled.li`
  margin: 0 !important;
  line-height: 34px;

  @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
    a {
      height: 22px;
    }
  }
`;

const TOC = ({ intl, sections }) => {
  const [active, setActive] = useState();

  if (!sections) return null;

  const titles = sections.filter(Boolean);

  if (!titles.length) return <LoadingIndicator />;

  return (
    <>
      <SectionHeading marginCollapse className="no-print">
        {intl.formatMessage(messages.on_this_page)}
      </SectionHeading>

      <ul className="links no-print">
        {titles.map(section => {
          const title = intl.formatMessage(section);

          return (
            <Li key={title}>
              <Link
                href={`#${title}`}
                onClick={() => {
                  setActive(title);
                }}
                className={active === title ? 'has-focus' : null}
                label={title}
              />
            </Li>
          );
        })}
      </ul>
    </>
  );
};

TOC.propTypes = {
  intl: intlShape.isRequired,
  sections: PropTypes.arrayOf(PropTypes.shape({})),
};

export default TOC;
