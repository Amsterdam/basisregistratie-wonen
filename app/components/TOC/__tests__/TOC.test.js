import React from 'react';
import { render, cleanup, fireEvent, waitForDomChange } from 'react-testing-library';
import 'jest-styled-components';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from '@datapunt/asc-ui';

import messages from '../../../translations/nl.json';
import TOC from '..';

const intlProvider = new IntlProvider({ locale: 'nl', messages });
const { intl } = intlProvider.getChildContext();

const sectionData = [
  {
    id: 'registraties.public_space',
  },
  {
    id: 'registraties.residence',
  },
  {
    id: 'registraties.number_identification',
  },
  {
    id: 'registraties.accommodation_object',
  },
  {
    id: 'registraties.house',
  },
  {
    id: 'registraties.cadastral_object',
  },
  null,
  {
    id: 'registraties.non_natural_person',
  },
  null,
  {
    id: 'registraties.area',
  },
];

let container;
let rerender;
let queryByTestId;

describe('TOC', () => {
  beforeEach(() => {
    ({ container, rerender, queryByTestId } = render(
      <ThemeProvider>
        <TOC intl={intl} sections={sectionData} />
      </ThemeProvider>,
    ));
  });
  afterEach(cleanup);

  it('matches the snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('should filter out empty values', () => {
    const numNonEmptyItems = sectionData.filter(Boolean).length;

    expect(container.getElementsByTagName('li')).toHaveLength(numNonEmptyItems);
  });

  it('should not print anything', () => {
    const immediateChildren = container.childNodes;
    immediateChildren.forEach(child => {
      expect(child.classList).toContain('no-print');
    });
  });

  it('should render null', () => {
    rerender(
      <ThemeProvider>
        <TOC intl={intl} />
      </ThemeProvider>,
    );

    expect(container.firstChild).toBeNull();
    expect(queryByTestId('progress-wrapper')).toBeNull();

    rerender(
      <ThemeProvider>
        <TOC intl={intl} sections={[]} />
      </ThemeProvider>,
    );

    expect(container.firstChild).not.toBeNull();
    expect(queryByTestId('progress-wrapper')).not.toBeNull();
  });

  it('should keep active element state', async () => {
    const anchors = document.getElementsByTagName('a');
    const randomAnchor = anchors[Math.floor(Math.random() * anchors.length)];

    Array.from(anchors).forEach(anchor => {
      expect(anchor.classList).not.toContain('has-focus');
    });

    fireEvent(
      randomAnchor,
      new MouseEvent('click', {
        bubbles: true,
      }),
    );
    waitForDomChange({ container }).then(() => {
      Array.from(anchors).forEach(anchor => {
        if (anchor === randomAnchor) {
          expect(anchor.classList).toContain('has-focus');
        } else {
          expect(anchor.classList).not.toContain('has-focus');
        }
      });
    });
  });
});
