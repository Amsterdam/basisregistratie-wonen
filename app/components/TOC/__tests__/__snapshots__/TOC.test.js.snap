// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOC matches the snapshot 1`] = `
.c0 {
  margin-top: 30px;
  margin-top: 0;
}

.c2 {
  padding: 0 !important;
}

.c2:focus {
  outline: 4px solid #ffc90e !important;
}

.c2:hover,
.c2:focus {
  color: #ec0000;
  fill: #ec0000;
}

.c4 {
  vertical-align: baseline;
  margin-right: 5px;
}

.c1 {
  margin: 0 !important;
  line-height: 34px;
}

.c3 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3:focus {
  outline: none;
  background-color: #FEC813;
}

.c3:hover {
  color: #ec0000;
}

.c5 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c5:focus {
  outline: none;
  background-color: #FEC813;
}

.c5:hover {
  color: #ec0000;
}

.c6 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6:focus {
  outline: none;
  background-color: #FEC813;
}

.c6:hover {
  color: #ec0000;
}

.c7 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c7:focus {
  outline: none;
  background-color: #FEC813;
}

.c7:hover {
  color: #ec0000;
}

.c8 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c8:focus {
  outline: none;
  background-color: #FEC813;
}

.c8:hover {
  color: #ec0000;
}

.c9 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c9:focus {
  outline: none;
  background-color: #FEC813;
}

.c9:hover {
  color: #ec0000;
}

.c10 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c10:focus {
  outline: none;
  background-color: #FEC813;
}

.c10:hover {
  color: #ec0000;
}

.c11 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c11:focus {
  outline: none;
  background-color: #FEC813;
}

.c11:hover {
  color: #ec0000;
}

@media print {
  .c0 {
    margin-top: 0;
  }
}

@media screen and (-ms-high-contrast:active),(-ms-high-contrast:none) {
  .c1 a {
    height: 22px;
  }
}

<div>
  <h3
    class="no-print c0"
  >
    Op deze pagina
  </h3>
  <ul
    class="links no-print"
  >
    <li
      class="c1"
    >
      <a
        class="c2 c3"
        href="#Openbare ruimte"
      >
        <img
          class="c4"
          width="14"
        />
        <span
          class="linklabel"
        >
          Openbare ruimte
        </span>
      </a>
    </li>
    <li
      class="c1"
    >
      <a
        class="c2 c5"
        href="#Woonplaats"
      >
        <img
          class="c4"
          width="14"
        />
        <span
          class="linklabel"
        >
          Woonplaats
        </span>
      </a>
    </li>
    <li
      class="c1"
    >
      <a
        class="c2 c6"
        href="#Nummeraanduiding"
      >
        <img
          class="c4"
          width="14"
        />
        <span
          class="linklabel"
        >
          Nummeraanduiding
        </span>
      </a>
    </li>
    <li
      class="c1"
    >
      <a
        class="c2 c7"
        href="#Verblijfsobject"
      >
        <img
          class="c4"
          width="14"
        />
        <span
          class="linklabel"
        >
          Verblijfsobject
        </span>
      </a>
    </li>
    <li
      class="c1"
    >
      <a
        class="c2 c8"
        href="#Pand"
      >
        <img
          class="c4"
          width="14"
        />
        <span
          class="linklabel"
        >
          Pand
        </span>
      </a>
    </li>
    <li
      class="c1"
    >
      <a
        class="c2 c9"
        href="#Kadastraal object"
      >
        <img
          class="c4"
          width="14"
        />
        <span
          class="linklabel"
        >
          Kadastraal object
        </span>
      </a>
    </li>
    <li
      class="c1"
    >
      <a
        class="c2 c10"
        href="#Kadastraal subject NNP"
      >
        <img
          class="c4"
          width="14"
        />
        <span
          class="linklabel"
        >
          Kadastraal subject NNP
        </span>
      </a>
    </li>
    <li
      class="c1"
    >
      <a
        class="c2 c11"
        href="#Gebied"
      >
        <img
          class="c4"
          width="14"
        />
        <span
          class="linklabel"
        >
          Gebied
        </span>
      </a>
    </li>
  </ul>
</div>
`;
